Sources/Baggage/BaggageContext.swift

```diff
@@ -132,7 +139,7 @@ public struct AnyBaggageContextKey {
         return self._name ?? String(describing: self.keyType.self)
     }
 
-    public init<Key>(_ keyType: Key.Type) where Key: BaggageContextKey {
+    init<Key>(_ keyType: Key.Type) where Key: BaggageContextKey {
```

Comment:
There's no need to have this be publicly initializable, right? Maybe we can even make it private and have a method on `BaggageContextKey` like `eraseToAnyBaggageContextKey()`, similar to Combines `eraseToAnyPublisher()`.